### ISSUE TITLE
Photo Upload App - Completion Items & Snag List 

### ISSUE THREAD

**Issue by barneyday** [HBHWoolacotts/RPii#150]
- [ ] Issues sending photos off to RPii - (Steve currently aware) - having to go into Administration and manually upload the queue.  Needs to be altered so it keeps checking and sending

- [x] Can we please add the Images icon into the Service Job Enquiry screen in RPii?  This is so service jobs can see the photos that have been uploaded.  I would suggest here:
![image](https://user-images.githubusercontent.com/10868496/45043659-ad458500-b065-11e8-8de6-586d774884a9.png)

- [x] Can you please add the files uploaded to the Documents Tab in the Service screen (obviously for service jobs only!).  That way engineers can see them too.
![image](https://user-images.githubusercontent.com/10868496/45043966-9eab9d80-b066-11e8-9c74-2b2ad1c08966.png)

- [ ] NEW FEATURE REQUEST - can we please add the ability to add TEXT to photos? @IrregularShed

**Comment by IrregularShed** [comment]
There's two parts to the app problem:

* it looks like there's issues with the handling the database on the server. Requests are being logged when attempting to access routes (literally the first line of each function) but dying somewhere after. The only common factor is accessing the SQLite database.
* the app isn't automatically retrying to send the saved jobs after the next successful 'live' upload, which is something I scribbled down in notes but apparently never got a chance to implement.

At the moment I'm changing the app to use [Knex](https://knexjs.org/) instead of raw SQLite to talk to the database. Once that's done I'll be in a position to swap to MySQL (which is on the server already) to see if that improves things (and if it doesn't, it at least comes with better logging).

**Comment by IrregularShed** [comment]
Right, I've got a version of the code that works using Knex (although still using the old DB). Starting to put in the MySQL bindings.

**Comment by IrregularShed** [comment]
The server software is now running on MySQL, and I've imported all the records from the old database. Hopefully that will result in a happier, more stable server (it's hard to tell without it being used properly yet but it seems to be from testing).

I'll look into getting saved uploads retriggered automatically in the morning - in the meantime @barneyday, if one or two guys with the app can try uploading the saved list and do a couple more jobs with it, that will provide massively beneficial data for me to nail any lingering issues.

**Comment by barneyday** [comment]
I haven't been made aware of any further issues.  How are the logs looking?

**Comment by IrregularShed** [comment]
All looking good - nothing scary in there! I'd say the server side of things is significantly improved. Currently juggling a couple of other things, so I've not had a chance to make headway with changes to the app, but it's not fallen off my radar.

**Comment by barneyday** [comment]
Issues uploading: 20/09/2018: I have sent an email to managers asking for feedback on the uploads.
St Austell have already replied and said it seems OK.  Will post further replies when I get them.

**Comment by gazmond** [comment]
Second and third points are done. You'll need to hard refresh as the javascript nd flash have changed.
